agent/src/beerocks/slave/beerocks_slave_main.cpp

```diff
@@ -391,6 +388,9 @@ static int run_son_slave(int slave_num, beerocks::config_file::sConfigSlave &bee
     son::slave_thread::sSlaveConfig son_slave_conf;
     fill_son_slave_config(beerocks_slave_conf, son_slave_conf, slave_num);
 
+    // read the number of failures allowed before stopping agent from platform configuration
+    son_slave_conf.stop_on_failure_attempts = bpl_cfg_get_stop_on_failure_attempts();
```

Comment:
Again, it's not possible to access bpl here. We're in a different process even (this is the `beerocks_agent -i <iface>` process, not the plain agent process).

So I think we have to initialize stop_on_failure_attempts to 0 and get it from platform_manager as part of cACTION_PLATFORM_SON_SLAVE_REGISTER_RESPONSE instead.

Reply:
Done. You are probably aware that if a timeout occurs while in STATE_WAIT_FOR_PLATFORM_MANAGER_REGISTER_RESPONSE, then thread will never stop but retry indefinitely because of that initial value that never gets updated.